arbor/lif_cell_group.cpp

```diff
@@ -123,11 +127,15 @@ void lif_cell_group::advance_cell(time_type tfinal, time_type dt, cell_gid_type
     std::size_t n_values = 0;
     std::vector<std::pair<time_type, sampler_association_handle>> samples;
     if (!samplers_.empty()) {
+        auto tlast = last_time_sampled_[lid];
         std::lock_guard<std::mutex> guard(sampler_mex_);
         for (auto& [hdl, assoc]: samplers_) {
-            if (assoc.probeset_ids.empty()) continue; // No need to generate events
-            // Construct sampling times
-            const auto& times = util::make_range(assoc.sched.events(t, tfinal));
+             // No need to generate events
+            if (assoc.probeset_ids.empty()) continue;
+            // Construct sampling times, might give us the last time we sampled, so skip that.
+            auto times = util::make_range(assoc.sched.events(tlast, tfinal));
+            if (!times.empty() && times.front() == tlast) times.left++;
```

Comment:
can we be sure that `tlast` doesn't appear twice or more times?

Reply:
Good point.

Reply:
...and fixed.